design_doc.md

```diff
@@ -5,14 +5,16 @@ Xarrays extension for DGGS. Technical specifications.
 ## Goals
 
 The goal of the `xdggs` library is to facilitate working with multiple Discrete Global Grid Systems (DGGSs) via a unified, high-level and user-friendly API that is deeply integrated with [Xarray](https://xarray.dev).
+This document describes the in-memory representation of DGGS data in Python environments.
 
 Examples of common DGGS features that `xdggs` should provide or facilitate:
 
 - convert a DGGS from/to another grid (e.g., a DGGS, a latitude/longitude rectilinear grid, a raster grid, an unstructured mesh)
 - convert a DGGS from/to vector data (points, lines, polygons, envelopes)
+- nearest neighbor search and bounding box queries around a given cell
```

Comment:
Aren't these special cases of selection by geometries (already mentioned below)? For example, in [xvec](https://github.com/xarray-contrib/xvec) you can achieve that on vector data cubes using `ds.xvec.query()` with a `shapely.bbox` object or an array of `shapely.Point` objects.

Reply:
Yes. However, bounding boxes are convex polygons so we can use much faster algorithms for this special case.